<a href="https://colab.research.google.com/github/apledepla/SA2_AFUNDAR_CUERDO_RODILLAS/blob/task1/TASK_1_AFUNDAR%2C_CUERDO%2C_RODILLASRDO%2C_RODILLAS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# From Data to Diagnosis: Exploring Diabetes Predicotrs through Bootstrap, Jackknife, and Inference Models

In [1]:
library(readr)
library(ggplot2)
library(dplyr)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




## Introduction


Diabetes mellitus is a chronic and potentially life-threatening condition that affects millions of people worldwide. It is characterized by the body's inability to produce or effectively use insulin, which leads to elevated blood glucose levels. Early detection of diabetes is crucial to preventing serious health complications such as heart disease, kidney failure, and neuropathy.


The rise of data-driven healthcare has enabled researchers to develop predictive models that can assist in early diagnosis. With the help of research techniques and inferential statistics, this project aims to explore the relationships between medical indicators and diabetes risk to develop more robust, accurate, and interpretable insights.

This study uses the Pima Indians Diabetes Dataset, a well-known dataset in the medical and machine learning communities, in order to investigate which clinical features are most informative for predicting the onset of diabetes.

## Statement of the Problem

Identifying the most significant predictors of diabetes remains a challenge due to variable distribution, sampling bias, and model overfitting. Thus, this project aims to determine which health-related variables most significantly predict the presence of diabetes in patiences, using statistical methods such as bootstrapping, jackknife resampling, permutation testing, and Bayesian inference.


Particularly, it will try to answer the following questions:

1.

## Exploratory Data Analysis

To better understand the structure and quality of the dataset, it is best to conduct a simple exploratory data analysis, to ensure  the nature of each variable is inspected before applying statistical inference techniques in later components.

### Overview

In [3]:
df <- read.csv("diabetes.csv")
head(df)

Pregnancies Glucose BloodPressure SkinThickness Insulin BMI 
1 6           148     72            35              0     33.6
2 1            85     66            29              0     26.6
3 8           183     64             0              0     23.3
4 1            89     66            23             94     28.1
5 0           137     40            35            168     43.1
6 5           116     74             0              0     25.6
  DiabetesPedigreeFunction Age Outcome
1 0.627                    50  1      
2 0.351                    31  0      
3 0.672                    32  1      
4 0.167                    21  0      
5 2.288                    33  1      
6 0.201                    30  0

The dataset contains 768 observations and 9 variables, including 8 predictor features and 1 binary outcome variable (Outcome), where:

- 0 indicates a non-diabetic patient

- 1 indicates a diabetic patient

### Variable Summary


In [4]:
summary(df)

  Pregnancies        Glucose      BloodPressure    SkinThickness  
 Min.   : 0.000   Min.   :  0.0   Min.   :  0.00   Min.   : 0.00  
 1st Qu.: 1.000   1st Qu.: 99.0   1st Qu.: 62.00   1st Qu.: 0.00  
 Median : 3.000   Median :117.0   Median : 72.00   Median :23.00  
 Mean   : 3.845   Mean   :120.9   Mean   : 69.11   Mean   :20.54  
 3rd Qu.: 6.000   3rd Qu.:140.2   3rd Qu.: 80.00   3rd Qu.:32.00  
 Max.   :17.000   Max.   :199.0   Max.   :122.00   Max.   :99.00  
    Insulin           BMI        DiabetesPedigreeFunction      Age       
 Min.   :  0.0   Min.   : 0.00   Min.   :0.0780           Min.   :21.00  
 1st Qu.:  0.0   1st Qu.:27.30   1st Qu.:0.2437           1st Qu.:24.00  
 Median : 30.5   Median :32.00   Median :0.3725           Median :29.00  
 Mean   : 79.8   Mean   :31.99   Mean   :0.4719           Mean   :33.24  
 3rd Qu.:127.2   3rd Qu.:36.60   3rd Qu.:0.6262           3rd Qu.:41.00  
 Max.   :846.0   Max.   :67.10   Max.   :2.4200           Max.   :81.00  
    Outcome  


From the summary, the following findings are observed:
- Glucose, BMI, and Age have reasonable ranges, but may not be normally distributed.

- Insulin and SkinThickness contain many zeros, which may represent missing values rather than true measurements.

- Outcome is binary and imbalanced (approx. 65% non-diabetic, 35% diabetic).